tests/web3js/eth_revert_reason_test.js

```diff
@@ -0,0 +1,95 @@
+const { assert } = require('chai')
+const conf = require('./config')
+const helpers = require('./helpers')
+const web3 = conf.web3
+
+it('store revertReason field in transaction receipts', async () => {
+    let deployed = await helpers.deployContract("storage")
+    let contractAddress = deployed.receipt.contractAddress
+
+    // make sure deploy was successful
+    assert.equal(deployed.receipt.status, conf.successStatus)
+
+    // assert that the receipt for the contract deployment transaction
+    // does not have any revert reason.
+    // Note: `revertReason` field is dropped from the result when
+    // using `web3.eth.getTransactionReceipt`, that's why we need
+    // to use `helpers.callRPCMethod`.
+    let receipt = await helpers.callRPCMethod(
+        'eth_getTransactionReceipt',
+        [deployed.receipt.transactionHash]
+    )
+    assert.isUndefined(receipt.body['result'].revertReason)
+
+    // we construct a transaction that reverts on purpose, with an assertion error
+    let callAssertError = deployed.contract.methods.assertError().encodeABI()
+    let assertErrorTx = {
+        from: conf.eoa.address,
+        to: contractAddress,
+        data: callAssertError,
+        gas: 1_000_000,
+        gasPrice: 0
+    }
+    let signedTx = await conf.eoa.signTransaction(assertErrorTx)
+
+    // we need to use `helpers.callRPCMethod` to test this out,
+    // because `web3.eth.sendSignedTransaction` will not let
+    // through transactions that revert, it will error out.
```

Comment:
I wonder why this is? is this expected? should we change it?

Reply:
Yeah, it is expected, not much we can do about it.

Under the hood, `web3.eth.sendSignedTransaction` will perform:
1. `eth_estimateGas`
2. `eth_call` (using as `gasLimit` what was returned from `eth_estimateGas`)
3. If `eth_call` is successful, it will then perform `eth_sendRawTransaction`

But `eth_call` fails, because it reverts, so it never proceeds with `eth_sendRawTransaction`, so the reverted transaction never gets indexed.